mlir/lib/Dialect/EmitC/Transforms/TypeConversions.cpp

```diff
@@ -56,9 +56,9 @@ std::optional<Type> mlir::emitc::getSignedTypeFor(Type ty) {
   if (ty.isInteger())
     return IntegerType::get(ty.getContext(), ty.getIntOrFloatBitWidth(),
                             IntegerType::SignednessSemantics::Signed);
-  if (isa<emitc::SizeTType>(ty))
-    return emitc::PtrDiffTType::get(ty.getContext());
-  if (isSupportedIntegerType(ty))
+  if (isa<SizeTType, SignedSizeTType>(ty))
```

Comment:
Why do we need to convert `SignedSizeTType` here - it's already signed?

Reply:
It's signed, but it does not (per spec, though in practice it does) allow arithmetic ops on negative values.